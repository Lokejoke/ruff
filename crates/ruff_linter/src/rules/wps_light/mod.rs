//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
