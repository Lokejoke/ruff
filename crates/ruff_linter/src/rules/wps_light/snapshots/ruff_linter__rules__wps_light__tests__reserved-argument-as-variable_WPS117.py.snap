---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
snapshot_kind: text
---
WPS117.py:2:14: WPS117 Reserved argument name self used as variable
  |
2 | def x1(a, b, self): ...
  |              ^^^^ WPS117
3 | 
4 | def x2(a, b, c): 
  |

WPS117.py:5:5: WPS117 Reserved argument name cls used as variable
  |
4 | def x2(a, b, c): 
5 |     cls = 3
  |     ^^^ WPS117
6 | 
7 | class X:
  |

WPS117.py:8:5: WPS117 Reserved argument name self used as variable
   |
 7 | class X:
 8 |     self = 3
   |     ^^^^ WPS117
 9 |     cls = 3
10 |     mcs = 3
   |

WPS117.py:9:5: WPS117 Reserved argument name cls used as variable
   |
 7 | class X:
 8 |     self = 3
 9 |     cls = 3
   |     ^^^ WPS117
10 |     mcs = 3
   |

WPS117.py:10:5: WPS117 Reserved argument name mcs used as variable
   |
 8 |     self = 3
 9 |     cls = 3
10 |     mcs = 3
   |     ^^^ WPS117
11 | 
12 |     def x3(a, b, self): ...
   |

WPS117.py:12:18: WPS117 Reserved argument name self used as variable
   |
10 |     mcs = 3
11 | 
12 |     def x3(a, b, self): ...
   |                  ^^^^ WPS117
13 | 
14 |     def x4(a, b, c): 
   |

WPS117.py:15:9: WPS117 Reserved argument name cls used as variable
   |
14 |     def x4(a, b, c): 
15 |         cls = 3
   |         ^^^ WPS117
16 | 
17 |     def x4(self=3): 
   |

WPS117.py:17:12: WPS117 Reserved argument name self used as variable
   |
15 |         cls = 3
16 | 
17 |     def x4(self=3): 
   |            ^^^^ WPS117
18 |         ...
   |

WPS117.py:20:5: WPS117 Reserved argument name cls used as variable
   |
18 |         ...
19 | 
20 | x = cls = 5
   |     ^^^ WPS117
21 | x = lambda self: self + 12
22 | self = 2
   |

WPS117.py:21:12: WPS117 Reserved argument name self used as variable
   |
20 | x = cls = 5
21 | x = lambda self: self + 12
   |            ^^^^ WPS117
22 | self = 2
23 | mcs = 2
   |

WPS117.py:22:1: WPS117 Reserved argument name self used as variable
   |
20 | x = cls = 5
21 | x = lambda self: self + 12
22 | self = 2
   | ^^^^ WPS117
23 | mcs = 2
   |

WPS117.py:23:1: WPS117 Reserved argument name mcs used as variable
   |
21 | x = lambda self: self + 12
22 | self = 2
23 | mcs = 2
   | ^^^ WPS117
   |
